### PR TITLE
Changed setting the state of the transction to COMMIT_FAILED rather than ROLLING_BACK upon commit failure

### LINKED ISSUE #8483: Client transaction timeout makes transaction not able to rollback
Commit failure on clients sets the transaction state to ROLLING_BACK, which makes the transaction non-rollbackable. This is not a problem for lock-based data structures (map, multimap) but it is problematic for collections. Since Hazelcast does not have an `auto-release the resources` for collections yet (planned for 3.8) one should call rollback explicitly upon commit failure in order to release resources.

Please see https://groups.google.com/d/msgid/hazelcast/26611d6a-4c00-4fac-91eb-0803bdb77cf6%40googlegroups.com?utm_medium=email&utm_source=footer

### PR DESCRIPTION
**Pull request by gurbuzali**
fixes #8483

### CODE REVIEW

File: hazelcast-client/src/test/java/com/hazelcast/client/txn/ClientTxnTest.java [hunk 1]
```diff
@@ -144,8 +146,27 @@ public void memberRemoved(MembershipEvent membershipEvent) {
...
+        IQueue<Object> queue = client.getQueue(name);
+        queue.offer(randomString());
+
+        TransactionOptions options = new TransactionOptions().setTimeout(2, TimeUnit.SECONDS);
+        TransactionContext context = client.newTransactionContext(options);
+        context.beginTransaction();
+        try {
+            context.getQueue(name).take();
+            sleepAtLeastSeconds(3);
+            context.commitTransaction();
```
Comment (ihsandemir) [inline hazelcast-client/src/test/java/com/hazelcast/client/txn/ClientTxnTest.java]: May be we can put fail() check after this line to make sure it is not executed for sure. Similarly for the XA test.

Comment (gurbuzali) [inline hazelcast-client/src/test/java/com/hazelcast/client/txn/ClientTxnTest.java]: 👍

**Comment by metanet** [comment]
👍

**Comment by eminn** [comment]
👍

**Pull request merged**